lib/rules/lines-around-comment.js

```diff
@@ -0,0 +1,144 @@
+    options.afterLineComment = options.afterLineComment || false;
+    options.beforeBlockComment = options.beforeBlockComment || false;
+    options.afterBlockComment = typeof options.afterBlockComment !== "undefined" ? options.afterBlockComment : true;
+
+    // Cache line numbers of comments for faster lookup
+    var COMMENTS;
+    var EMPTY_LINES;
+    var ALL_LINES;
+
+    /**
+     * Determines if a line number is either empty or contains a comment.
+     * @param {number} lineNumber The line that we should check for being an empty.
+     * @returns {boolean} True if the lineNumber is either empty OR a comment.
+     */
+    function isEmptyOrComment(lineNumber) {
+        return COMMENTS.indexOf(lineNumber) > -1 || EMPTY_LINES.indexOf(lineNumber) > -1;
+    }
+
+    /**
+     * Returns whether or not comments are not on lines starting with or ending with code
+     * @param {ASTNode} node The comment node to check.
+     * @returns {boolean} True if the comment is not alone.
+     */
+    function codeAroundComment(node) {
+
+        // Get the whole line and cut it off at the start of the comment
+        var startLine = String(ALL_LINES[node.loc.start.line - 1]);
+        var endLine = String(ALL_LINES[node.loc.end.line - 1]);
+
+        var preamble = startLine.slice(0, node.loc.start.column).trim();
+
+        // Also check after the comment
+        var postamble = endLine.slice(node.loc.end.column).trim();
+
+        // Should be false if there was only whitespace around the comment
+        return !!(preamble || postamble);
+    }
+
+    /**
+     * Checks if a comment node has lines around it (ignores inline comments)
+     * @param {ASTNode} node The Comment node.
+     * @param {Object} opts Options to determine the newline.
+     * @param {Boolean} opts.after Should have a newline after this line.
+     * @param {Boolean} opts.before Should have a newline before this line.
+     * @returns {void}
+     */
+    function checkForEmptyLine(node, opts) {
+
+        var after = opts.after,
+            before = opts.before;
+
+        var prevLineNum = node.loc.start.line - 1,
+            nextLineNum = node.loc.end.line + 1,
+            commentIsNotAlone = codeAroundComment(node);
+
+        // we ignore all inline comments
+        if (commentIsNotAlone) {
+            return;
+        }
+
+        // check for newline before
+        if (before && !isEmptyOrComment(prevLineNum)) {
+            context.report(node, "Expected line before comment.");
+        }
+
+        // check for newline after
+        if (after && !isEmptyOrComment(nextLineNum)) {
+            context.report(node, "Expected line after comment.");
+        }
+
+    }
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    return {
+
+        "Program": function() {
+            COMMENTS = getAllComments(context);
```

Comment:
This was weird but seemed to work. Not sure if caching these matters or helps at all.